internal/oci/oci.go

```diff
@@ -302,19 +302,21 @@ func (r *Runtime) StopContainer(ctx context.Context, c *Container, timeout int64
 
 // DeleteContainer deletes a container.
 func (r *Runtime) DeleteContainer(c *Container) (err error) {
-	impl, err := r.RuntimeImpl(c)
-	if err != nil {
-		return err
+	r.runtimeImplMapMutex.RLock()
+	impl, ok := r.runtimeImplMap[c.ID()]
+	r.runtimeImplMapMutex.RUnlock()
+	if !ok {
+		impl, _ = r.newRuntimeImpl(c)
```

Comment:
Hm, what are we going to do with that error?

Reply:
Check added on the error return.